.github/workflows/sphinx-build.yml

```diff
@@ -11,7 +10,25 @@ on:
     - main
 
 jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install -r requirements.txt
+    - name: unittest sample code 
+      run: |
+        python -m unittest discover -s .\test\ -p "*_test.py"
+
   sphinx:
+    needs:
+    - test
```

Comment:
Sphinx build doesn't really need unit tests to pass. I'd move this to be a dependency of `deploy`, which will have the added benefit of being executed in parallel then.